core/packages/contracts/src/NativeTokensTypes.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.19;
+
+import "./ScaleCodec.sol";
+import "./SubstrateTypes.sol";
+
+library NativeTokensTypes {
```

Comment:
Haven't properly reviewed this file, nor the corresponding ScaleCode changes, SubstrateTypes changes or Bytes/Memory changes, since my knowledge of Substrate types and encodings is low - should make sure to get an additional review/audit of these files from someone more familiar with those bits.